docs/src/conventions.md

```diff
@@ -0,0 +1,116 @@
+# Coding Conventions
+
+## Formatting
+
+We automate most of our formatting preferences. Our CI will run format checkers for PRs and pushes.
+These checks are required for merging any PR.
+
+For rust, we use [rustfmt](https://github.com/rust-lang/rustfmt)
+which is configured via [rustfmt.toml](https://github.com/model-checking/kani/blob/main/rustfmt.toml) file.
+We are also in the process of enabling clippy.
+Because of that, we still have a couple of lints disabled (see [.cargo/config](https://github.com/model-checking/kani/blob/main/.cargo/config.toml) for the updated list).
+
+We also have a bit of C code and python in our repository.
+For C we use clang-format and for python scripts we use autopep8.
+See [.clang-format](https://githubcom/model-checking/kani/blob/main/.clang-format)
+and [pyproject.toml](https://github.com/model-checking/kani/blob/main/scripts/pyproject.toml)
+for their configuration.
+
+
+### Exceptions
+
+We recognize that in some cases, the formatting and lints automation may not be applicable to a specific code.
+In those cases, we usually prefer explicitly allowing exceptions by locally disabling the check.
+E.g., use `#[allow]` annotation instead of disabling a link on a crate or project level.
+
+### Copyright notice
+
+All source code files begin with a copyright and license notice. If this is a new file, please add the following notice:
+
+```rust
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+```
+
+When modifying a file from another project, please keep their headers as is and append the following notice after them:
+
+```rust
+// ... existing licensing headers ...
+
+// Modifications Copyright Kani Contributors
+// See GitHub history for details.
+```
+
+We also have automated checks for the copyright notice.
+There are a few file types where this rules doesn't apply.
+You can see that list in the [copyright-exclude](
+https://github.com/model-checking/kani/blob/main/scripts/ci/copyright-exclude) file.
+
```

Comment:
Add something along the lines of: "Non-Rust files (`.sh`, `.py`, `.toml`) use the same header except the comments start with `# `.